### PR TITLE
Drops env dump from JSON output, introduces System.env wrapper

### LINKED ISSUE #4: Drop env dump from JSON output
data-validator may expose secrets held in environment variables in the output JSON.

https://github.com/target/data-validator/blob/67eec1d599828909ef21e715d5c4ad5cf180ee4a/src/main/scala/com/target/data_validator/ValidatorConfig.scala#L97-L100 dumps the current environment into the output JSON via https://github.com/target/data-validator/blob/67eec1d599828909ef21e715d5c4ad5cf180ee4a/src/main/scala/com/target/data_validator/ValidatorConfig.scala#L74 that calls https://github.com/target/data-validator/blob/67eec1d599828909ef21e715d5c4ad5cf180ee4a/src/main/scala/com/target/data_validator/ValidatorConfig.scala#L102 which includes it here https://github.com/target/data-validator/blob/67eec1d599828909ef21e715d5c4ad5cf180ee4a/src/main/scala/com/target/data_validator/ValidatorConfig.scala#L114

It's safe to dump variables that data-validator accesses but it's unwise to dump everything.

### PR DESCRIPTION
**Pull request by colindean**
Fixes #4

~WIP because it needs tests and the tests that I've written emit some wild errors that seem related to Spark access~. Edit Oct 22: All implemented tests pass and no longer emit errors

### CODE REVIEW

File: src/main/scala/com/target/data_validator/EnvironmentVariables.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    def toString: String
+  }
+  case class Present(value: String) extends EnvVarResult {
+    override def toString: String = value
+  }
+  case class Inaccessible(message: String) extends EnvVarResult {
+    override val toString: String = s"<inaccessible: $message>"
+  }
+  object Inaccessible {
+    def apply(throwable: Throwable): Inaccessible = Inaccessible(throwable.getMessage)
```
Comment (colindean) [inline src/main/scala/com/target/data_validator/EnvironmentVariables.scala]: Only thing I dislike about this is the discard of the throwable. However, there are so few conditions under which System.getenv throws an exception that its message would likely suffice for troubleshooting.

**Comment by colindean** [comment]
Wait until #26 is merged to merge this.

**Comment by colindean** [comment]
#26 is merged, so this can be reviewed for real now.

File: src/main/scala/com/target/data_validator/EnvironmentVariables.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package com.target.data_validator
+
+import scala.collection.mutable
+import scala.util.Try
+
+object EnvironmentVariables {
+  type MaybeEnvVar = Try[Option[String]]
+
+  val accessedEnvVars: mutable.Map[String, MaybeEnvVar] = mutable.Map.empty
+  private[this] def clearAccessList(): Unit = accessedEnvVars.clear()
```
Comment (phpisciuneri) [inline src/main/scala/com/target/data_validator/EnvironmentVariables.scala]: no usage

```suggestion
```

File: src/main/scala/com/target/data_validator/EnvironmentVariables.scala [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      whenPresent = { Present }
+    )
+      .recover { case throwable: Throwable => Error(throwable) }
+      .get
+  }
+
+  def getWithHandlers[T](key: String)(whenError: PartialFunction[Throwable, T],
+                                      whenUnset: => T,
+                                      whenPresent: String => T
+  ): Try[T] = {
```
Comment (phpisciuneri) [inline src/main/scala/com/target/data_validator/EnvironmentVariables.scala]: Can you indent this consistent with the rest of the code base?  Put first arg on a new line and use 2 space indenting.

File: src/main/scala/com/target/data_validator/EnvironmentVariables.scala [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    getWithHandlers(key)(
+      whenError = { case throwable: Throwable => Inaccessible(throwable) },
+      whenUnset = { Unset },
+      whenPresent = { Present }
+    )
+      .recover { case throwable: Throwable => Error(throwable) }
+      .get
+  }
+
+  def getWithHandlers[T](key: String)(whenError: PartialFunction[Throwable, T],
```
Comment (phpisciuneri) [inline src/main/scala/com/target/data_validator/EnvironmentVariables.scala]:
```suggestion
  def getWithHandlers[T](key: String)(
    whenError: PartialFunction[Throwable, T],
    whenUnset: => T,
    whenPresent: String => T
```

File: src/main/scala/com/target/data_validator/EnvironmentVariables.scala [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                                      whenUnset: => T,
```
Comment (phpisciuneri) [inline src/main/scala/com/target/data_validator/EnvironmentVariables.scala]:
```suggestion
```

File: src/main/scala/com/target/data_validator/EnvironmentVariables.scala [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                                      whenPresent: String => T
```
Comment (phpisciuneri) [inline src/main/scala/com/target/data_validator/EnvironmentVariables.scala]:
```suggestion
```

File: src/main/scala/com/target/data_validator/EnvironmentVariables.scala (hunk 4 above)
Comment (phpisciuneri) [reply src/main/scala/com/target/data_validator/EnvironmentVariables.scala]: > Can you indent this consistent with the rest of the code base? Put first arg on a new line and use 2 space indenting.

**Comment by colindean** [comment]
> Did you see my comment about it not working as I would have expected?

Yeah. I've not gotten around to double-checking. It's been a while since I did this work 😉

**Comment by colindean** [comment]
This is working as expected now. I ran DV 0.11.0 against the test data and ran it again with this patchset. The former output every envvar while the latter output only the envvars touched.

**Comment by phpisciuneri** [comment]
@colindean Tested in a few different environments, looks good to me.

Can you remove `mylog.properties` before merging?  Looks like it might have been accidentally commit.

**Pull request merged**